### PR TITLE
Endpoint branch

### PR DESCRIPTION
**Pull request by Ghezz9**

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -1,35 +1,212 @@
...
+    }
+
+    jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
+        if (err) {
+            return res.status(403).json({ error: 'Invalid token.' });
+        }
+        req.user = user;
+        next();
+    });
+}
```
Comment (DanielWilliamParsons) [inline server/server.js:30]: Nice! Great to see this in use. We will need to figure out how to send the token to the front end after we authorize the user.

File: server/server.js [hunk 2]
```diff
@@ -1,35 +1,212 @@
...
+
+        const token = jwt.sign({ username: user.username }, process.env.ACCESS_TOKEN_SECRET, { expiresIn: '1h' });
+        res.cookie('token', token, { httpOnly: true, secure: true, maxAge: 3600000 });
+
+        return res.status(200).json({ message: 'Login successful' });
+    } catch (error) {
+        return res.status(500).json({ error: 'Error during login' });
+    }
+});
+
```
Comment (DanielWilliamParsons) [inline server/server.js]: Oh 😮 I'm a little surprised to see this.

You asked me to work on authorization this morning, so I opened an issue and I spent the morning writing the login and the register functionality. I told you I was almost finished by lunch. Then after lunch I tried to ask you if we could combine our code, but you said we were wasting time talking.

Sorry to say, I feel like I wasted a whole morning writing the authorization part if you already wrote this and I didn't know about it 😕.

I think we need to make the division of our tasks clearer in the morning and stop rushing into coding.

File: src/db/src/db/data/seeds/001favorites.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+exports.seed = async function(knex) {
+  await knex('favorite').del();
+
+  await knex('favorite').insert([
+    { user_id: 1, recipe_id: 'spaghetti_bolognese', region: 'Emilia-Romagna' },
+    { user_id: 2, recipe_id: 'carbonara', region: 'Lazio' }
+  ]);
+};
```
Comment (DanielWilliamParsons) [inline src/db/src/db/data/seeds/001favorites.js:8]: 😲 Oh! You made a database and inserted data, too. Why did you ask me to work on the database if you planned to do it anyway? I asked you to talk about how to seed the database. You even approved my pull request. So, I am very confused.

I think we may have some communication problems in our group that we should address in the morning. Right now, I feel that all the work I have done yesterday afternoon and today has not provided a single contribution to our group.

**Comment by Ghezz9** [comment]
It's your database Daniel, Gustavo at some moment of the day said to me to
pull your database branch for trying the first test, i'm so sorry if you
feel like this, my bad comunication is the problem😔 tomorrow morning we
can all togetheter the test and see what happens, i'm writing also in the
parrot channel but maybe there isn't a notification when a people write on
that channel. I think you didn't wasted your time😁

Il Mer 23 Ott 2024, 21:03 Daniel William Parsons ***@***.***>
ha scritto:

**Review by DanielWilliamParsons — APPROVED** [review]
Davide, I'm sorry to say that I think we may have some communication problems in our group. Please see my comments. I feel that I have contributed nothing to this project so far and I have wasted a day and a half.

I'm approving these changes because they allow us to move on with the project. My contribution is unnecessary because you have done everything you asked me to do.

Please can we make our division of tasks clearer from now on - as the tech lead, this is your responsibility. In addition, let's please make sure that we spend some time talking about our code. Regularly updating each other on questions, issues and next-steps can go a long way to avoiding communication problems.

**Review by jacksontfm — APPROVED** [review]
Reviewed

**Pull request merged**